src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -0,0 +1,1841 @@
+ 			System.out.println(clk.getName());
+ 			if(GlobalSignalRouting.crRoutes != null || GlobalSignalRouting.dstINTtileRoutes != null) {
+ 				if(GlobalSignalRouting.dstINTtileRoutes == null) {
+ 					System.out.println("INFOR: Route with clock skew data reference");
+ 					GlobalSignalRouting.clkRouteWithClkSkewRouteDelays(clk, this.design.getDevice());
+ 				}else {
+ 					System.out.println("INFO: Route with clock route and timing data");
+ 					GlobalSignalRouting.clkEnableRoute(clk, this.design.getDevice());
+ 				}
+ 			}else {
+ 				if(this.config.isSymmetricClkRouting()) {
+ 					System.out.println("INFO: Route with symmetric non-timing-driven clock router");
+ 	 				GlobalSignalRouting.symmetricClkRouting(clk, this.design.getDevice());
+ 				}else {
+ 					System.out.println("INFO: Route with default non-timing-driven clock router");
+ 	 				GlobalSignalRouting.defaultClkRoute(clk, this.design.getDevice());
+ 				}
+ 			}
+			this.preserveNet(clk);
+ 		}
+	}
+	
+	/**
+	 * Adds and initialize a regular signal net to the list of routing targets.
+	 * @param net The net to be added for routing.
+	 * @param multiSLR A flag to indicate if the device has multiple SLRs, for the sake of avoiding unnecessary check for single-SLR devices.
+	 */
+	protected void addNetConnectionToRoutingTargets(Net net, boolean multiSLR) {
+		net.unroute();
+		this.numWireNetsToRoute++;;
+		this.createsNetWrapperAndConnections(net, this.config.getBoundingBoxExtensionX(), this.config.getBoundingBoxExtensionY(),multiSLR);
+	}
+	
+	/**
+	 * Adds a static net to the static net routing target list.
+	 * @param staticNet The static net in question, i.e. VCC or GND.
+	 */
+	protected void addStaticNetRoutingTargets(Net staticNet){
+		List<SitePinInst> sinks = new ArrayList<>();
+		for(SitePinInst sink : staticNet.getPins()){
+			if(sink.isOutPin()) continue;
+			sinks.add(sink);
+		}
+		
+		if(sinks.size() > 0 ) {
+			for(SitePinInst sink : sinks) {
+				this.addReservedNode(sink.getConnectedNode(), staticNet);
+			}
+			this.staticNetAndRoutingTargets.put(staticNet, sinks);
+		}else {
+			this.preserveNet(staticNet);
+			this.numNotNeedingRoutingNets++;	
+		}
+	}
+	
+	protected void addStaticNetRoutingTargets(Net staticNet, List<SitePinInst> sinks) {
+		this.staticNetAndRoutingTargets.put(staticNet, sinks);
+	}
+	
+	/**
+	 * Routes static nets with preserved resources list supplied to avoid conflicting nodes.
+	 * @param preservedConnectedNodesToPins The list of preserved nodes for other nets.
+	 */
+	private void routeStaticNets(List<Node> preservedConnectedNodesToPins){
+		GlobalSignalRouting.setDesignRoutethruHelper(this.design, this.routethruHelper);
+		
+		for(Net net : this.staticNetAndRoutingTargets.keySet()){
+			for(SitePinInst sink : this.staticNetAndRoutingTargets.get(net)) {
+				this.preservedNodes.remove(sink.getConnectedNode());
+			}
+		}
+		
+		RouterHelper.invertPossibleGndPinsToVccPins(this.design, this.design.getGndNet());
```

Comment:
Why inverting from GND to VCC desirable here?

Reply:
It is because of the hardware. There are invertible BELs related to those pins. Without this inverting, Vivado would complain about the routing of GND and VCC.